### DISCUSSION TITLE
is `sudo` really required for npm-install-ing pyright globally on MacOS?

### DISCUSSION THREAD

**Discussion by frou** [microsoft/pyright#3832] [General]
The pyright README says:

> On MacOS or Linux, sudo is required to install globally: sudo npm install -g pyright

However at least from the MacOS perspective, that strikes me as odd. We don't have to use `sudo` when npm-install-ing any other such tools globally. And sure enough `pyright` also seems to install and run fine without it.

**Comment by erictraut** [comment]
Moving to discussion since this isn't a bug report.

**Reply by erictraut** [reply]
In my experience, sudo is required on MacOS. Perhaps it's only on older versions of node. I'll change the documentation to say "may".

**Reply by frou** [reply]
Perhaps there's a difference based on whether Node itself was installed using Homebrew (that's what I do), versus using the installer from the Node website. It's not a big deal anyway - I just mentioned it because in my experience, unnecessary use of `sudo` can cause other permission-related problems down the line.

**Reply by erictraut** [reply]
Yeah, that might explain it. I typically use the installer from the Node website, so that's consistent with your theory.
